client/src/components/HistoryViewer/HistoryViewer.js

```diff
@@ -109,10 +145,16 @@ class HistoryViewer extends Component {
       ':version': currentVersion,
     };
 
+    const version = this.getVersions()
+      .filter((eachVersion) => eachVersion.Version === currentVersion)[0];
+    const latestVersion = this.getLatestVersion();
```

Comment:
Use `find` to avoid looping through the whole array.

```js
const version = this.getVersions().find(version => version.Version === currentVersion);
```